src/module/hook/user/index.ts

```diff
@@ -121,6 +121,12 @@ export function useCreateUser() {
         }
       },
       onSuccess: async (data, newUser) => {
+        // Prevent flash of default data while refetching
+        queryClient.setQueryData<IUser[]>(
```

Comment:
Here's the main fix. The rest of the PR was just cleaning up the user creation form.